storage/filesystem/dotgit/dotgit.go

```diff
@@ -946,6 +949,14 @@ func (d *DotGit) addRefFromHEAD(refs *[]*plumbing.Reference) error {
 
 func (d *DotGit) readReferenceFile(path, name string) (ref *plumbing.Reference, err error) {
 	path = d.fs.Join(path, d.fs.Join(strings.Split(name, "/")...))
+	st, err := d.fs.Stat(path)
```

Comment:
Might be better to open first and then `fstat()`, to avoid going through path lookup twice?  Also prevents TOCTOU type issues.  Maybe not a big deal.

Reply:
My initial patch was to attempt an `f.Stat()` after open, but it turns out the `File` involved here isn't an `os.File` but part of this osfs astraction layer from $otherModule, and it doesn't offer an equivalent that will effectively `fstat(File)` in a platform-independent way. =-(